src/Proto.OpenTelemetry/OpenTelemetryDecorators.cs

```diff
@@ -0,0 +1,207 @@
+﻿using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using OpenTelemetry.Trace;
+
+namespace Proto.OpenTelemetry
+{
+    class OpenTelemetryRootContextDecorator : RootContextDecorator
+    {
+        private readonly ActivitySetup _sendActivitySetup;
+
+        public OpenTelemetryRootContextDecorator(IRootContext context, ActivitySetup sendActivitySetup) : base(context)
+            => _sendActivitySetup = (activity, message)
+                => {
+                activity?.SetTag(ProtoTags.ActorType, "<None>");
+                sendActivitySetup(activity, message);
+            };
+
+        public override void Send(PID target, object message)
+            => OpenTelemetryMethodsDecorators.Send(target, message, _sendActivitySetup, () => base.Send(target, message));
+
+        public override void Request(PID target, object message)
+            => OpenTelemetryMethodsDecorators.Request(target, message, _sendActivitySetup, () => base.Request(target, message));
+
+        public override void Request(PID target, object message, PID? sender)
+            => OpenTelemetryMethodsDecorators.Request(target, message, sender, _sendActivitySetup, () => base.Request(target, message, sender));
+
+        public override Task<T> RequestAsync<T>(PID target, object message, CancellationToken cancellationToken)
+            => OpenTelemetryMethodsDecorators.RequestAsync(target, message, _sendActivitySetup,
+                () => base.RequestAsync<T>(target, message, cancellationToken)
+            );
+    }
+
+    class OpenTelemetryActorContextDecorator : ActorContextDecorator
+    {
+        private readonly ActivitySetup _receiveActivitySetup;
+        private readonly ActivitySetup _sendActivitySetup;
+
+        public OpenTelemetryActorContextDecorator(
+            IContext context,
+            ActivitySetup sendActivitySetup,
+            ActivitySetup receiveActivitySetup
+        ) : base(context)
+        {
+            var actorType = context.Actor.GetType().Name;
+            var self = context.Self.ToString();
+            _sendActivitySetup = (activity, message) => {
+                activity?.SetTag(ProtoTags.ActorType, actorType);
+                activity?.SetTag(ProtoTags.ActorPID, self);
```

Comment:
I'm not 100% sure we need ActorPID here. The PID is already present in the trace, but depending on the span type, it is either  SenderPid or TargetPid. ActorPid is only to normalize this to a single value